### ISSUE TITLE
Named markers

### ISSUE THREAD

**Issue by turicas** [turicas/templater#5]
We should be able to name markers and if we do, `Templater.parser` wil return a `dict` instead of a `list`.
The named markers should be enabled if `marker` parameter is a regular expression object (returned by `re.compile`). The regular expression marker must have one group with the name of the marker.
An example code:

``` python
from re import compile as re_compile
from templater import Templater


template = Templater(template='{{first}}<b>{{second}}</b>{{third}}',
                     marker=re_compile(r'{{([a-z]*)}}'))
print template.parse('testing <b>named</b> markers')
```

It should print:

```
{'second': 'named', 'third': ' markers', 'first': 'testing '}
```

> Note: if more than one marker have the same name, the key in returned `dict` should be overwritten by the last occurrence.

**Comment by turicas** [comment]
`Templater.learn` can create/delete some holes. What to do if the template is using named markers?

Possible solutions:
- Split the learning and parsing process. Create classes `TemplateLearner` (which will have the methods `learn`, `save` and `dump`) and `TemplateParser` (which will have the methods `parse`, `open` and `load`).
  - _Cons_: inability to learn from a pre-created template (even if it does not use named markers).
- Simply doesn't have the feature of learning with templates which have named markers. Should raise an exception if the user call the `learn` method in a `Templater` object that has named markers.

**Comment by turicas** [comment]
Issue implemented in branch `feature/named-markers` (was merged into master and released in version 0.4.0).
